File: Basics to advance working with DataGridView Checkbox columns/[C#][VB]-Basics to advance working with DataGridView Checkbox columns/C# and VB/DoubleDualing/Extensions/BindingSourceExtensions.vb
Module BindingSourceExtensions
    <System.Diagnostics.DebuggerStepThrough()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Function DataTable(ByVal sender As BindingSource) As DataTable
        If sender.DataSource.GetType.Equals(GetType(System.Data.DataTable)) Then
            Return DirectCast(sender.DataSource, DataTable)
        Else
            Throw New Exception("DataSource is not a DataTable")
        End If
    End Function

    <System.Diagnostics.DebuggerStepThrough()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Function CurrentRow(ByVal sender As BindingSource) As DataRow
        Return CType((CType(sender.Current, DataRowView)).Row, DataRow)
    End Function


End Module

File: Basics to advance working with DataGridView Checkbox columns/[C#][VB]-Basics to advance working with DataGridView Checkbox columns/C# and VB/CheckOneRow_C/Extensions/BindingSourceExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using System.Data;
using System.Windows.Forms;

static class BindingSourceExtensions
{
    public static string CurrentRow(this BindingSource sender, string Column)
    {
        return ((DataRowView)sender.Current).Row[Column].ToString();
    }
}
